packages/@uppy/utils/src/getAllowedMetaFields.ts

```diff
@@ -0,0 +1,14 @@
+import type { Meta } from './UppyFile'
+
+export default function getAllowedMetaFields<M extends Meta>(
+  fields: string[] | boolean,
+  meta: M,
+): string[] {
+  if (Array.isArray(fields)) {
+    return fields
+  }
+  if (fields === true) {
+    return Object.keys(meta)
+  }
```

Comment:
nit: I would start by the `true` comparison, as I assume it's cheaper than `isArray` call
```suggestion
  if (fields === true) {
    return Object.keys(meta)
  }
  if (Array.isArray(fields)) {
    return fields
  }
```